packages/shared/apis/RestrictedActions.ts

```diff
@@ -62,10 +59,7 @@ export class RestrictedActions extends ExposableAPI {
   @exposeMethod
   async triggerEmote(emote: Emote): Promise<void> {
     // checks permissions
-    if (!this.hasPermission(PermissionItem.ALLOW_TO_TRIGGER_AVATAR_EMOTE)) {
-      defaultLogger.error(`Permission "${PermissionItem.ALLOW_TO_TRIGGER_AVATAR_EMOTE}" is required`)
-      return
-    }
+    await this.ensureHasPermissions([PermissionItem.ALLOW_TO_TRIGGER_AVATAR_EMOTE])
```

Comment:
idiomatically `ensure` returns a sanitized or initializes the needed thing
and `require` or `assert` is the protocol assertion